### PR TITLE
[Docs] Adds tutorial and stepwise instructions for EXTERN

### PR DESCRIPTION
**Pull request by techdocsmith**
- Adds a tutorial for EXTERN to export data with MSQ
-  Fixes up the style of the EXTERN function reference
This PR has:

- [ x] been self-reviewed.

### CODE REVIEW

File: docs/tutorials/tutorial-extern.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+```
+
+### Start Druid and load sample data
+
+From the root of the Druid distribution, launch Druid as follows:
+
+```bash
+./bin/start-druid
+```
+
```
Comment (ektravel) [inline docs/tutorials/tutorial-extern.md:76]: Open http://localhost:8888/ in your browser to access the web console.

File: docs/tutorials/tutorial-extern.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+  "metroCode",
+  "countryIsoCode",
+  "regionName"
+FROM "ext"
+PARTITIONED BY DAY
+```
+
+### Query to export data
+
+Run the following query to export query results to the path:
```
Comment (ektravel) [inline docs/tutorials/tutorial-extern.md:121]:
```suggestion
Open a new tab and run the following query to export query results to the path:
```

**Review by ektravel — COMMENTED** [review]
Initial feedback, still testing the steps.

File: docs/multi-stage-query/reference.md [hunk 3]
```diff
@@ -216,11 +223,11 @@ FROM <table>
 
 Supported arguments to the function:
```
Comment (ektravel) [inline docs/multi-stage-query/reference.md:224]:
```suggestion
Supported arguments for the function:
```

File: docs/multi-stage-query/reference.md [hunk 4]
```diff
@@ -216,11 +223,11 @@ FROM <table>
...
 
-| Parameter   | Required | Description                                                                                                                                                                                                                                              | Default |
-|-------------|--------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------| --|
-| `exportPath`  | Yes | Absolute path to a subdirectory of `druid.export.storage.baseDir` used as the destination to export the results to. The export query expects the destination to be empty. If the location includes other files or directories, then the query will fail. | n/a |
+| Parameter | Required | Description | Default |
+|---|---|---|---|---|
```
Comment (ektravel) [inline docs/multi-stage-query/reference.md:227]:
```suggestion
|---|---|---|---|
```

File: docs/multi-stage-query/reference.md [hunk 5]
```diff
@@ -146,25 +151,26 @@ FROM <table>
...
-| `druid.export.storage.s3.maxRetry`           | No       | Defines the max number times to attempt S3 API calls to avoid failures due to transient errors.                                                                                                                                      | 10  |
-| `druid.export.storage.s3.chunkSize`          | No       | Defines the size of each chunk to temporarily store in `tempDir`. The chunk size must be between 5 MiB and 5 GiB. A large chunk size reduces the API calls to S3, however it requires more disk space to store the temporary chunks. | 100MiB |
+| Runtime Parameter | Required | Description | Default |
+|---|---|---|---|
+| `druid.export.storage.s3.allowedExportPaths` | Yes | Array of S3 prefixes allowed as export destinations. Export queries fail if the export destination does not match any of the configured prefixes. For eample: `[\"s3://bucket1/export/\", \"s3://bucket2/export/\"]` | n/a |
```
Comment (ektravel) [inline docs/multi-stage-query/reference.md:163]:
```suggestion
| `druid.export.storage.s3.allowedExportPaths` | Yes | Array of S3 prefixes allowed as export destinations. Export queries fail if the export destination does not match any of the configured prefixes. For example: `[\"s3://bucket1/export/\", \"s3://bucket2/export/\"]` | n/a |
```

File: docs/multi-stage-query/reference.md [hunk 6]
```diff
@@ -179,29 +185,30 @@ FROM <table>
...
+|---|---|---|---|
+| `bucket`    | Yes | GCS bucket destination for exported files. You must add the bucket and prefix combination to the `druid.export.storage.google.allowedExportPaths` allow list. | n/a |
+| `prefix` | Yes  | Destination path in the bucket to create exported files. The export query expects the destination path to be empty. If the location includes other files, the query will fail. You must add the bucket and prefix combination to the `druid.export.storage.google.allowedExportPaths` allow list. | n/a |
+
+Configure the following runtime parameters to export query results to a GCS destination:
 
-The following runtime parameters must be configured to export into a GCS destination:
+| Runtime Parameter | Required | Description | Default |
+|---|---|---|---|
+| `druid.export.storage.google.allowedExportPaths` | Yes | Array of GCS prefixes allowed as export destinations. Export queries fail if the export destination does not match any of the configured prefixes. For eample: `[\"gs://bucket1/export/\", \"gs://bucket2/export/\"]` | n/a     |
```
Comment (ektravel) [inline docs/multi-stage-query/reference.md:197]:
```suggestion
| `druid.export.storage.google.allowedExportPaths` | Yes | Array of GCS prefixes allowed as export destinations. Export queries fail if the export destination does not match any of the configured prefixes. For example: `[\"gs://bucket1/export/\", \"gs://bucket2/export/\"]` | n/a     |
```

**Review by ektravel — COMMENTED** [review]
I tested the steps for the local file system.

**Comment by ektravel** [comment]
Not for this PR, but it would be good to standardize how we refer to SQL functions in this topic. Specifically when it comes to using code font.

File: docs/multi-stage-query/reference.md [hunk 7]
```diff
@@ -95,12 +94,14 @@ For more information, see [Read external data with EXTERN](concepts.md#read-exte
 
 #### `EXTERN` to export to a destination
 
-`EXTERN` can be used to specify a destination where you want to export data to.
-This variation of EXTERN requires one argument, the details of the destination as specified below.
-This variation additionally requires an `AS` clause to specify the format of the exported rows.
+You can use `EXTERN` to specify a destination to export data.
+This variation of `EXTERN` accepts the details of the destination as the only argument and requires an `AS` clause to specify the format of the exported rows.
+- accepts the details of the destination as the only argument
+- requires an `AS` clause to specify the format of the exported rows.
```
Comment (techdocsmith) [inline docs/multi-stage-query/reference.md:100]:
```suggestion
```

File: docs/multi-stage-query/reference.md [hunk 8]
```diff
@@ -95,12 +94,14 @@ For more information, see [Read external data with EXTERN](concepts.md#read-exte
...
+
+When you export data, Druid creates metadata files in a subdirectory of the destination directory named `_symlink_format_manifest`.
```
Comment (techdocsmith) [inline docs/multi-stage-query/reference.md:102]:
```suggestion
When you export data, Druid creates metadata files in a subdirectory named `_symlink_format_manifest`.
```

File: docs/multi-stage-query/reference.md [hunk 9]
```diff
@@ -128,10 +127,14 @@ SELECT
 FROM <table>
 ```
 
-##### S3
+##### S3 - Amazon S3
+
+To export results to S3, pass the `s3()` function as an argument to the `EXTERN` function.
+Export to S3 requires the `druid-s3-extensions` extension
```
Comment (ektravel) [inline docs/multi-stage-query/reference.md:133]:
```suggestion
Export to S3 requires the `druid-s3-extensions` extension.
```

File: docs/multi-stage-query/reference.md [hunk 10]
```diff
@@ -146,25 +149,26 @@ FROM <table>
...
-| `prefix`    | Yes      | Path where the exported files would be created. The export query expects the destination to be empty. If the location includes other files, then the query will fail. The bucket and prefix combination should be whitelisted in `druid.export.storage.s3.allowedExportPaths`. | n/a     |
+| Parameter | Required | Description | Default |
+|---|---|---|---|
+| `bucket` | Yes  | S3 bucket destination for exported files. You must add the bucket and prefix combination to the `druid.export.storage.s3.allowedExportPaths` allow list. | n/a |
+| `prefix` | Yes  | Destination path in the bucket to create exported files. The export query expects the destination path to be empty. If the location includes other files, the query will fail. You must add the bucket and prefix combination to the `druid.export.storage.s3.allowedExportPaths` allow list. | n/a |
 
-The following runtime parameters must be configured to export into an S3 destination:
+Configure following runtime parameters to export to an S3 destination:
```
Comment (ektravel) [inline docs/multi-stage-query/reference.md:157]:
```suggestion
Configure the following runtime parameters to export to an S3 destination:
```

**Review by ektravel — APPROVED** [review]
A couple of nits. Otherwise, LGTM.

File: docs/tutorials/tutorial-extern.md [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+## Export query results to cloud storage
+
+The steps to export to cloud storage are similar to exporting to the local file system.
+Druid supports Amazon S3 or Google Cloud Storage (GCS) as cloud storage destinations.
+
+1. Enable the extension for your cloud storage destination. See [Loading core extensions](../configuration/extensions.md#loading-core-extensions).
+   - **Amazon S3**: `druid-s3-extensions`
+   - **GCS**: `google-extensions`
+  See [Loading core extensions](../configuration/extensions.md#loading-core-extensions) for more information.
+1. Configure the additional properties for your cloud storage destination. Replace `{CLOUD}` with `s3` or `google` accordingly:
```
Comment (adarshsanjeev) [inline docs/tutorials/tutorial-extern.md:155]: The numbering seems off here and in other places in the file.

**Review by adarshsanjeev — APPROVED** [review]
Looks good to me overall!

File: docs/tutorials/tutorial-extern.md (hunk 11 above)
Comment (techdocsmith) [reply docs/tutorials/tutorial-extern.md:155]: Using 1. for all the numbers lets the build system maintain the numbers for easier editing. Markdown translates the numbers to `<ol><li / ><li /><li /></ol>`

Comment (adarshsanjeev) [reply docs/tutorials/tutorial-extern.md:155]: Ah, got it! Thanks for explaining!

**Pull request merged**